Provider crash when using config block in assets for cloudflare_workers_script

### Confirmation

- [x] This is a bug with an existing resource and is not a feature request or enhancement. Feature requests should be submitted with Cloudflare Support or your account team.
- [x] I have searched the issue tracker and my issue isn't already found.
- [x] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

OpenTofu: v1.9.1
Cloudflare provider: v5.9.0(also observed in v5.8.3)

### Affected resource(s)

- cloudflare_workers_script

### Terraform configuration files

```hcl
`resource "cloudflare_workers_script" "worker_deployment" {
  account_id         = var.cloudflare_account_id
  script_name        = var.worker_script_name
  main_module        = "main.ts"
  content            = <<EOT
export default {
  async fetch(request, env, ctx) {
    // ...existing code...
  },
};
EOT
  compatibility_date = "2024-07-29"
  bindings = concat(
    local.has_assets ? [{ name = "ASSETS", type = "assets" }] : []
  )
  logpush = true
  assets = local.has_assets ? {
    jwt = jwt
    config = {
      run_worker_first = true
      # other config keys...
    }
  } : null
}`
```

### Link to debug output

Error: Plugin did not respond

### Panic output

```
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-cloudflare_v5.8.3 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x106be46a4]

goroutine 9 [running]:
github.com/cloudflare/terraform-provider-cloudflare/internal/services/workers_script.(*WorkersScriptResource).Update(0x140004940b8, {0x109311f08, 0x140004971a0}, {{{{0x1093964d0, 0x1400003a360}, {0x108662460, 0x14000245380}}, {0x1093c5370, 0x140004acdc0}}, {{{0x1093964d0, ...}, ...}, ...}, ...}, ...)
        github.com/cloudflare/terraform-provider-cloudflare/internal/services/workers_script/resource.go:131 +0x1f4
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).UpdateResource(0x14000157688, {0x109311f08, 0x140004971a0}, 0x1400047f398, 0x1400047f368)
        github.com/hashicorp/terraform-plugin-framework@v1.15.0/internal/fwserver/server_updateresource.go:150 +0x768
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0x14000157688, {0x109311f08, 0x140004971a0}, 0x140000ee930, 0x1400047f5d8)
        github.com/hashicorp/terraform-plugin-framework@v1.15.0/internal/fwserver/server_applyresourcechange.go:118 +0x140
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0x14000157688, {0x109311f08?, 0x14000497080?}, 0x1400003ce10)
        github.com/hashicorp/terraform-plugin-framework@v1.15.0/internal/proto6server/server_applyresourcechange.go:71 +0x434
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x14000286960, {0x109311f08?, 0x14000496120?}, 0x140000ec100)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov6/tf6server/server.go:941 +0x2a8
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x10922a5e0, 0x14000286960}, {0x109311f08, 0x14000496120}, 0x14000263f00, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:687 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400020e200, {0x109311f08, 0x14000496060}, 0x140000d28a0, 0x14000378900, 0x10c95d908, 0x0)
        google.golang.org/grpc@v1.72.1/server.go:1405 +0xca8
google.golang.org/grpc.(*Server).handleStream(0x1400020e200, {0x109312808, 0x140004fe000}, 0x140000d28a0)
        google.golang.org/grpc@v1.72.1/server.go:1815 +0x910
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.72.1/server.go:1035 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 36
        google.golang.org/grpc@v1.72.1/server.go:1046 +0x13c

Error: The terraform-provider-cloudflare_v5.8.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected output

The resource should apply successfully and the config block should be accepted in the assets map.



### Actual output

The provider crashes with a segmentation fault when the config block is present in the assets map. If the config block is commented out, the resource applies successfully.



### Steps to reproduce

1. Create a `cloudflare_workers_script` resource with an `assets` map that includes a `config` block.
3. Run tofu apply (or terraform apply).
5. Provider crashes with nil pointer dereference.

### Additional factoids

- Commenting out the config block allows the resource to apply successfully.
- Observed in Cloudflare provider versions v5.8.3 and v5.9.0.

### References

_No response_

--- a/internal/services/workers_script/resource.go
+++ b/internal/services/workers_script/resource.go
@@ -66,6 +66,11 @@ func (r *WorkersScriptResource) Create(ctx context.Context, req resource.CreateR
 	var data *WorkersScriptModel
 
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
 	resp.Diagnostics.Append(req.Config.GetAttribute(ctx, path.Root("migrations"), &data.Migrations)...)
 
 	if resp.Diagnostics.HasError() {
@@ -128,6 +133,11 @@ func (r *WorkersScriptResource) Update(ctx context.Context, req resource.UpdateR
 	var data *WorkersScriptModel
 
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
 	resp.Diagnostics.Append(req.Config.GetAttribute(ctx, path.Root("migrations"), &data.Migrations)...)
 
 	if resp.Diagnostics.HasError() {

